docs/source/app_developers_guide/address_and_namespace.rst

```diff
@@ -0,0 +1,147 @@
+****************************
+Address and Namespace Design
+****************************
+
+
+Overview
+========
+
+Sawtooth Lake stores data within a Merkle Tree. Data is stored in leaf nodes,
+and each node is accessed using an addressing scheme that is composed of 35
+bytes, represented as 70 hex characters. The recommended way to construct an
+address is to use the hex-encoded hash values of the string or strings that
+make up the address elements. However, the encoding of the address is
+completely up to the transaction family defining the namespace, and does not
+need to involve hashing. Hashing is a useful way to deterministically generate
+likely non-colliding byte arrays of a fixed length.
+
+
+Address Components
+==================
+
+An address begins with a namespace prefix of six hex characters representing
+three bytes. The rest of the address, 32 bytes represented as 64 hex
+characters, can be calculated in various ways. However, certain guidelines
+should be followed when creating addresses, and specific requirements must be
+met.
+
+
+.. image:: ../images/address_namespace.*
+   :width: 100%
+   :align: center
+   :alt: Address and Namespace Design
+
+The address must be deterministic: that is, any validator or client that needs
+to calculate the address must be able to calculate the same address, every
+time, when given the same inputs.
+
+
+Namespace Prefix
+================
+
+All data under a namespace prefix follows a consistent address and data
+encoding/serialization scheme that is determined by the transaction family
+which defines the namespace.
+
+
+The namespace prefix consists of six hex characters, or three bytes.  An
+example namespace prefix that utilizes the string making up the transaction
+family namespace name to calculate the prefix is demonstrated by the following
+Python code:
+
+
+.. code-block:: python
+
+	prefix = hashlib.sha256("example_txn_family_namespace".encode('utf-8')).hexdigest()[:6]
+
+
+Alternatively, a namespace prefix can utilize an arbitrary scheme. The current
+configuration transaction family uses a prefix of ‘000000’, for example.
+
+
+Address Construction
+====================
+
+The rest of the address, or remaining 32 bytes (64 hex characters), must be
+calculated using a defined deterministic encoding format. Each address within
+a namespace should be unique, or the namespace consumers must be able to deal
+with collisions in a deterministic way.
+
+The addressing schema can be as simple or as complex as necessary, based on
+the requirements of the transaction family.
+
+Simple Example
+--------------
+
+This is a simple transaction family that stores key value pairs, e.g. “x=1”,
+or “y=2”, similar to the included intkey transaction family.
+
+The transaction family prefix is:
+
+.. code-block:: python
+
+	hashlib.sha256("intkey".encode('utf-8')).hexdigest()[:6]
```

Comment:
This says it is an example of a transaction family like intkey and uses a hash of 'intkey' to find the prefix, but intkey uses sha-512 hash instead of sha-256. This might confuse people. I suggest either the example doesn't reference intkey, and the prefix could be the hash of 'simple-example', or the hash algorithm for address construction that intkey uses, is used.